rust_icu_unum/src/lib.rs

```diff
@@ -263,12 +271,83 @@ impl UNumberFormat {
         format_double_currency_impl(
             versioned_function!(unum_formatDoubleCurrency),
             self.rep.as_ptr(),
-            number, 
+            number,
             currencyz.as_mut_c_ptr(),
             0 as *mut sys::UFieldPosition,
         )
     }
-   
+
+    /// Implements `unum_parseToUFormattable`. Since 0.3.1.
+    ///
+    /// > **WARNING** the `parse_position` parameter is with respect to the number index
+    /// in the `UChar` string.  This won't work exactly for multibyte UTF8 values of
+    /// `text`.  If you think you will have multibyte values, use instead
+    /// [UNumberFormat::parse_to_formattable_ustring].
+    pub fn parse_to_formattable<'a>(
+        &'a self,
+        text: &str,
+        parse_position: Option<i32>,
+    ) -> Result<uformattable::UFormattable<'a>, common::Error> {
+        let ustr = ustring::UChar::try_from(text)?;
+        self.parse_to_formattable_ustring(&ustr, parse_position)
+    }
+
+    /// Implements `unum_parseToUFormattable`. Since 0.3.1.
+    pub fn parse_to_formattable_ustring<'a>(
+        &'a self,
+        text: &ustring::UChar,
+        parse_position: Option<i32>,
+    ) -> Result<uformattable::UFormattable<'a>, common::Error> {
+        let mut fmt = uformattable::UFormattable::try_new()?;
+        let mut status = common::Error::OK_CODE;
+        let mut pos = parse_position.unwrap_or(0);
+        unsafe {
+            assert!(common::Error::is_ok(status));
+            versioned_function!(unum_parseToUFormattable)(
+                self.rep.as_ptr(),
+                fmt.as_mut_ptr(),
+                text.as_c_ptr(),
+                text.len() as i32,
+                &mut pos,
+                &mut status,
+            )
+        };
+        common::Error::ok_or_warning(status)?;
+        Ok(fmt)
+    }
+
+    /// Implements `unum_formatUFormattable`. Since 0.3.1.
+    pub fn format_formattable<'a>(
+        &self,
+        fmt: &uformattable::UFormattable<'a>,
+    ) -> Result<String, common::Error> {
+        let result = self.format_formattable_ustring(fmt)?;
+        String::try_from(&result)
+    }
+
+    /// Implements `unum_formatUFormattable`. Since 0.3.1.
+    pub fn format_formattable_ustring<'a>(
+        &self,
+        fmt: &uformattable::UFormattable<'a>,
+    ) -> Result<ustring::UChar, common::Error> {
+        const CAPACITY: usize = 200;
```

Comment:
Is this capacity recommended somewhere in ICU4C documentation? Consider mentioning the source in a comment.